tests/wandb_run_test.py

```diff
@@ -317,15 +317,21 @@ def test_use_artifact_offline(live_mock_server, test_settings):
     run.finish()
 
 
-def test_run_urls(test_settings):
+@pytest.mark.skip(
+    reason=(
+        "mock server enforces entity=mock_server_entity and project=test,"
+        "this should only work on a live server, or mock_server would need customization."
+    )
+)
```

Comment:
This used to work bc previously, a Settings object passed to `wandb.init()` would "carry over" its setting sources. settings.update(...) by default uses Source.OVERRIDE meaning that no downstream changes would affect the changed settings (unless again overridden). 
We could customize `mock_server` so that it "recognizes" two (or more) entities, but it feels a bit outside the scope for this PR.